api/v1beta1/metal3machine_webhook.go

```diff
@@ -53,28 +53,8 @@ func (c *Metal3Machine) ValidateDelete() error {
 
 func (c *Metal3Machine) validate() error {
 	var allErrs field.ErrorList
-	if c.Spec.Image.URL == "" {
-		allErrs = append(
-			allErrs,
-			field.Invalid(
-				field.NewPath("spec", "Image", "URL"),
-				c.Spec.Image.URL,
-				"is required",
-			),
-		)
-	}
 
-	// Checksum is not required for live-iso.
-	if c.Spec.Image.Checksum == "" && (c.Spec.Image.DiskFormat == nil || *c.Spec.Image.DiskFormat != LiveIsoDiskFormat) {
-		allErrs = append(
-			allErrs,
-			field.Invalid(
-				field.NewPath("spec", "Image", "Checksum"),
-				c.Spec.Image.Checksum,
-				"is required",
-			),
-		)
-	}
+	allErrs = append(allErrs, c.Spec.Image.Validate(*field.NewPath("Spec", "Image"))...)
```

Comment:
Side note: this assumes an Image is required. BMH can have a CustomDeploy instead.

Reply:
Thanks @dtantsur for pointing to CustomDeploy. This looks pretty interesting. However, for CAPM3 workflow Image field is still a required field. CAPM3 hasn't adopted yet the customdeploy workflow.

Also in BMO your implementation didnt land with any Documentation, and there is no mention of that here https://github.com/metal3-io/baremetal-operator/blob/main/docs/api.md#image.  I think it will be good to have some doc on this in BMO.